Composer/packages/client/src/pages/design/VisualEditorHeader.tsx

```diff
@@ -115,30 +116,37 @@ const useBreadcrumbs = (projectId: string, pluginConfig?: PluginConfig) => {
 
   if (encodedFocused) {
     // we've linked to an action, so put that in too
-    breadcrumbArray.push({
-      key: 'action-' + focusPath,
+    array.push({
+      key: `${BreadcrumbKeyPrefix.Action}-${focusPath}`,
       label: getActionName(possibleAction, pluginConfig),
     });
   }
 
   const currentDialog = (dialogs.find(({ id }) => id === dialogId) ?? dialogs[0]) as DialogInfo;
 
-  const selectedActions = useMemo(() => {
-    const actionSelected = Array.isArray(visualEditorSelection) && visualEditorSelection.length > 0;
-    if (!actionSelected) return [];
-
-    const selectedActions = visualEditorSelection.map((id) => get(currentDialog?.content, id));
-
-    return selectedActions;
-  }, [visualEditorSelection, currentDialog?.content]);
-
-  if (selectedActions.length === 1 && selectedActions[0] != null) {
-    const action = selectedActions[0] as any;
-    const actionName = getActionName(action, pluginConfig);
-
-    breadcrumbArray = [...breadcrumbArray.slice(0, 2), { key: 'action-' + actionName, label: actionName }];
-  }
-
+  // get newest label for breadcrumbs
+  const breadcrumbArray = useMemo(() => {
+    if (currentDialog.content) {
+      array.map((b) => {
+        const prefix = b.key.charAt(0);
+        const name = b.key.substr(2);
```

Comment:
I recommend that defining a pair of functions to parse / build the key, no matter how the key is parsed / built.

```
// Remember to make it typed
const buildKey = () => {};
const parseKey = () => {};
```

Reply:
use parse/build key function